src/main/java/com/mikahil/tarasevich/dao/impl/AbstractCrudDaoImpl.java

```diff
@@ -0,0 +1,129 @@
+        try {
+            preparedStatement.setInt(1, integer);
+            // preparedStatement.setObject();
+        } catch (SQLException e) {
+            //logger
+        }
+    };
+    protected static final BiConsumer<PreparedStatement, String> STRING_PARAM_SETTER = (preparedStatement, str) -> {
+        try {
+            preparedStatement.setString(1, str);
+        } catch (SQLException e) {
+            //logger
+        }
+    };
+    protected final ConnectorDB connector;
+    private final String findByIdQuery;
+    private final String findAllQuery;
+    private final String deleteByIdQuery;
+
+
+    protected AbstractCrudDaoImpl(ConnectorDB connector, String findByIdQuery, String findAllQuery, String deleteByIdQuery) {
+        this.connector = connector;
+        this.findByIdQuery = findByIdQuery;
+        this.findAllQuery = findAllQuery;
+        this.deleteByIdQuery = deleteByIdQuery;
+    }
+
+    @Override
+    public Optional<E> findById(Integer id) {
+        return findByParam(id, findByIdQuery, INT_PARAM_SETTER);
+    }
+
+    protected <P> Optional<E> findByParam(P param, String findByParam, BiConsumer<PreparedStatement,P> designatedParamSetter) {
+        try (final PreparedStatement preparedStatement =
+                     connector.getConnection().prepareStatement(findByParam)) {
+            designatedParamSetter.accept(preparedStatement, param);
+            try (final ResultSet resultSet = preparedStatement.executeQuery()) {
+                if (resultSet.next()) {
+                    return Optional.of(mapResultSetToEntity(resultSet));
+                }
+            }
+        } catch (SQLException e) {
+            //log
+            throw new DataBaseSqlRuntimeException("", e);
+        }
+        return Optional.empty();
+    }
+
+    public Optional<List<E>> findAll() {
+        try (final PreparedStatement preparedStatement =
+                     connector.getConnection().prepareStatement(findAllQuery)) {
+            return collectFoundEntitiesToList(preparedStatement);
+        } catch (SQLException e) {
+            //log
+            throw new DataBaseSqlRuntimeException("", e);
+        }
+    }
+
+    @Override
+    public void update(E entity) {
+
+    }
+
+    @Override
+    public void deleteById(Integer id) {
+        deleteById(id, INT_PARAM_SETTER);
+    }
+
+    private void deleteById(Integer id, BiConsumer<PreparedStatement,Integer> designatedParamSetter) {
+        try (final PreparedStatement preparedStatement =
+                     connector.getConnection().prepareStatement(deleteByIdQuery)) {
+            designatedParamSetter.accept(preparedStatement, id);
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            //log
+            throw new DataBaseSqlRuntimeException("", e);
+        }
+    }
+
+    protected Optional<List<E>> collectFoundEntitiesToList(PreparedStatement preparedStatement){
```

Comment:
make private and use findManyByStringParam findManyByIntParam and should return List<E>